presto-phoenix/src/main/java/com/facebook/presto/plugin/phoenix/PhoenixClient.java

```diff
@@ -0,0 +1,678 @@
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.RealType.REAL;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TimeType.TIME;
+import static com.facebook.presto.spi.type.TimeWithTimeZoneType.TIME_WITH_TIME_ZONE;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.type.TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.createUnboundedVarcharType;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.lang.Math.min;
+import static java.util.Collections.nCopies;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static org.apache.hadoop.hbase.HConstants.FOREVER;
+import static org.apache.phoenix.util.PhoenixRuntime.getTable;
+
+public class PhoenixClient
+{
+    private static final Logger log = Logger.get(PhoenixClient.class);
+
+    private static final Map<Type, String> SQL_TYPES = ImmutableMap.<Type, String>builder()
+            .put(BOOLEAN, "BOOLEAN")
+            .put(BIGINT, "BIGINT")
+            .put(INTEGER, "INTEGER")
+            .put(SMALLINT, "SMALLINT")
+            .put(TINYINT, "TINYINT")
+            .put(DOUBLE, "DOUBLE")
+            .put(REAL, "FLOAT")
+            .put(VARBINARY, "VARBINARY")
+            .put(DATE, "DATE")
+            .put(TIME, "TIME")
+            .put(TIME_WITH_TIME_ZONE, "TIME")
+            .put(TIMESTAMP, "TIMESTAMP")
+            .put(TIMESTAMP_WITH_TIME_ZONE, "TIMESTAMP")
+            .build();
+
+    protected final String connectorId;
+    protected final Driver driver = new PhoenixDriver();
+    protected final String connectionUrl;
+    protected final Properties connectionProperties;
+
+    @Inject
+    public PhoenixClient(PhoenixConnectorId connectorId, PhoenixConfig config) throws SQLException
+    {
+        this.connectorId = requireNonNull(connectorId, "connectorId is null").toString();
+
+        requireNonNull(config, "config is null");
+        connectionUrl = config.getConnectionUrl();
+        connectionProperties = new Properties();
+        connectionProperties.putAll(config.getConnectionProperties());
+    }
+
+    public Set<String> getSchemaNames()
+    {
+        try (PhoenixConnection connection = getConnection();
+                ResultSet resultSet = connection.getMetaData().getSchemas()) {
+            ImmutableSet.Builder<String> schemaNames = ImmutableSet.builder();
+            while (resultSet.next()) {
```

Comment:
why haven't you used `prest-jdbc-base` to implement this connector?

Reply:
I originally proceeded with the project by inheriting `` presto-base-jdbc``. But I had to inherit or modify most of the class files, and the variables of the existing classes became unnecessary variables. For example, `` ConnectorSplit`` related classes, `` PhoenixSplit`` and `` PhoenixPageSource``, are representative examples.